### PR TITLE
MEP_Engine: Transform added for Fitting objects

### LINKED ISSUE #2545: MEP_Engine: Add Modify.Transform method for Fitting objects
#### Description:
The transform method is required for certain operations when converting from Revit fitting objects, similar to Cable Trays and Ducts.

### PR DESCRIPTION
**Pull request by tiagogrossi**
### Issues addressed by this PR

Closes #2545 

The Fitting object requires a `Modify.Transform` method for certain Revit convert operations.

### Changelog
Modified `BH.Engine.MEP.Modify.Transform`
Modified `BH.Engine.MEP.Query.Geometry`

### Additional comments
Added a safety check in `BH.Engine.MEP.Query.Geometry`.

### CODE REVIEW

**Comment by tiagogrossi** [comment]
@BHoMBot check compliance

File: MEP_Engine/Modify/Transform.cs [hunk 1]
```diff
@@ -26,15 +26,39 @@
...
+        {
+            if (!transform.IsRigidTransformation(tolerance))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Transformation failed: only rigid body transformations are currently supported.");
+                return null;
+            }
+
+            Fitting result = fitting.GetShallowClone() as Fitting;
+            result.Location = result.Location.Transform(transform);
+            result.ConnectionsLocation = result.ConnectionsLocation.Select(x=>x.Transform(transform)).ToList();
```
Comment (FraserGreenroyd) [inline MEP_Engine/Modify/Transform.cs:55]:
```suggestion
            result.ConnectionsLocation = result.ConnectionsLocation.Select(x => x.Transform(transform)).ToList();
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Very minor formatting change requested.

**Comment by tiagogrossi** [comment]
@BHoMBot check core

**Comment by tiagogrossi** [comment]
@BHoMBot check required

**Comment by tiagogrossi** [comment]
@BHoMBot check dataset-compliance
@BHoMBot check copyright-compliance

**Comment by tiagogrossi** [comment]
Thanks for looking into this @FraserGreenroyd , I think it's ready for your approval and merge now.

File: MEP_Engine/Modify/Transform.cs [hunk 2]
```diff
@@ -26,15 +26,39 @@
...
+        [Output("transformed", "Modified Fitting with unchanged properties, but transformed location and connections location points.")]
+        public static Fitting Transform(this Fitting fitting, TransformMatrix transform, double tolerance = Tolerance.Distance)
+        {
+            if (!transform.IsRigidTransformation(tolerance))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Transformation failed: only rigid body transformations are currently supported.");
+                return null;
+            }
+
+            Fitting result = fitting.GetShallowClone() as Fitting;
```
Comment (FraserGreenroyd) [inline MEP_Engine/Modify/Transform.cs:53]:
```suggestion
            Fitting result = fitting.ShallowClone();
```

Based on the work @alelom is currently doing, it's probably best not to put in another `GetShallowClone` when he's removing them. @alelom can you confirm this is the right suggestion for correction based on your current work?

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

File: MEP_Engine/Modify/Transform.cs (hunk 2 above)
Comment (alelom) [reply MEP_Engine/Modify/Transform.cs:53]: Yes definitely! Please don't give me another one to remove! 🤣

**Comment by tiagogrossi** [comment]
@BHoMBot check required

**Comment by tiagogrossi** [comment]
@BHoMBot check required

**Comment by tiagogrossi** [comment]
Stale PR, closed for fresh attempt on the feature.

**Pull request closed without merging**